diff --git a/bcdiceCore.rb b/bcdiceCore.rb
index 4b75c97..00570df 100644
--- a/bcdiceCore.rb
+++ b/bcdiceCore.rb
@@ -869,7 +869,7 @@ class BCDice
     string, secret, count, swapMarker = getD66Infos(dice)
     unless  string.nil?
       value = getD66ValueByMarker(swapMarker)
-      diceText = (value / 10).to_s + "," + (value % 10).to_s
+      diceText = value.div(10).to_s + "," + (value % 10).to_s
       return value, diceText
     end
 
diff --git a/diceBot/Cthulhu7th.rb b/diceBot/Cthulhu7th.rb
index 1ecd6f0..356e308 100644
--- a/diceBot/Cthulhu7th.rb
+++ b/diceBot/Cthulhu7th.rb
@@ -342,7 +342,7 @@ INFO_MESSAGE_TEXT
   end
 
   def getSetOfBullet(diff)
-    bullet_set_count = diff / 10
+    bullet_set_count = diff.div(10)
 
     if ((diff >= 1) and (diff < 10))
       bullet_set_count = 1  #技能値が9以下での最低値保障処理
@@ -352,7 +352,7 @@ INFO_MESSAGE_TEXT
   end
 
   def getHitBulletCountBase(diff, bullet_set_count)
-    hit_bullet_count_base = (bullet_set_count / 2)
+    hit_bullet_count_base = bullet_set_count.div(2)
 
     if ((diff >= 1) and (diff < 10))
       hit_bullet_count_base = 1  #技能値9以下での最低値保障
diff --git a/diceBot/Cthulhu7th_ChineseTraditional.rb b/diceBot/Cthulhu7th_ChineseTraditional.rb
index cbaaf24..cba5efc 100644
--- a/diceBot/Cthulhu7th_ChineseTraditional.rb
+++ b/diceBot/Cthulhu7th_ChineseTraditional.rb
@@ -344,7 +344,7 @@ INFO_MESSAGE_TEXT
   end
 
   def getSetOfBullet(diff)
-    bullet_set_count = diff / 10
+    bullet_set_count = diff.div(10)
 
     if ((diff >= 1) and (diff < 10))
       bullet_set_count = 1  #技能值９以下的最低限度保障處理
@@ -354,7 +354,7 @@ INFO_MESSAGE_TEXT
   end
 
   def getHitBulletCountBase(diff, bullet_set_count)
-    hit_bullet_count_base = (bullet_set_count / 2)
+    hit_bullet_count_base = bullet_set_count.div(2)
 
     if ((diff >= 1) and (diff < 10))
       hit_bullet_count_base = 1  #技能值９以下的最低限度保障處理
diff --git a/diceBot/Cthulhu7th_Korean.rb b/diceBot/Cthulhu7th_Korean.rb
index 1ecd6f0..356e308 100644
--- a/diceBot/Cthulhu7th_Korean.rb
+++ b/diceBot/Cthulhu7th_Korean.rb
@@ -340,7 +340,7 @@ INFO_MESSAGE_TEXT
   end
 
   def getSetOfBullet(diff)
-    bullet_set_count = diff / 10
+    bullet_set_count = diff.div(10)
 
     if ((diff >= 1) and (diff < 10))
       bullet_set_count = 1  #기능 수치가 9 이하일 때의 최저수치 보장 처리
@@ -350,7 +350,7 @@ INFO_MESSAGE_TEXT
   end
 
   def getHitBulletCountBase(diff, bullet_set_count)
-    hit_bullet_count_base = (bullet_set_count / 2)
+    hit_bullet_count_base = bullet_set_count.div(2)
 
     if ((diff >= 1) and (diff < 10))
       hit_bullet_count_base = 1  #기능 수치가 9 이하일 때의 최저수치 보장
diff --git a/diceBot/CthulhuTech.rb b/diceBot/CthulhuTech.rb
index d7e0c09..cbed04a 100644
--- a/diceBot/CthulhuTech.rb
+++ b/diceBot/CthulhuTech.rb
@@ -36,7 +36,7 @@ INFO_MESSAGE_TEXT
   end
 
   def check_nD10_nomalTest(total_n, dice_n, signOfInequality, diff, dice_cnt, dice_max, n1, n_max)
-    if n1 >= (dice_cnt / 2 + 0.9).to_i
+    if n1 >= (dice_cnt.div(2) + 0.9).floor
       return " ＞ ファンブル"
     end
 
diff --git a/diceBot/DiceBot.rb b/diceBot/DiceBot.rb
index 44385af..aeb377f 100644
--- a/diceBot/DiceBot.rb
+++ b/diceBot/DiceBot.rb
@@ -477,13 +477,13 @@ class DiceBot
         item, value = get_table_by_d66(table)
         value = value.to_i
         output = item[1]
-        diceText = (value / 10).to_s + "," + (value % 10).to_s
+        diceText = value.div(10).to_s + "," + (value % 10).to_s
         [output, value, diceText]
       when 'D66S'
         table = getTableInfoFromExtraTableText(table, 21)
         output, value = get_table_by_d66_swap(table)
         value = value.to_i
-        diceText = (value / 10).to_s + "," + (value % 10).to_s
+        diceText = value.div(10).to_s + "," + (value % 10).to_s
         [output, value, diceText]
       else
         raise "invalid dice Type #{command}"
diff --git a/diceBot/Dracurouge.rb b/diceBot/Dracurouge.rb
index 215d7f4..ec12728 100644
--- a/diceBot/Dracurouge.rb
+++ b/diceBot/Dracurouge.rb
@@ -85,7 +85,7 @@ MESSAGETEXT
     oneCount = countTargetDice(diceList, 1)
     sixCount = countTargetDice(diceList, 6)
 
-    gloryDiceCount = (oneCount / 2) + (sixCount / 2)
+    gloryDiceCount = oneCount.div(2) + sixCount.div(2)
     return gloryDiceCount
   end
 
diff --git a/diceBot/Dracurouge_Korean.rb b/diceBot/Dracurouge_Korean.rb
index f1a967d..464e12c 100644
--- a/diceBot/Dracurouge_Korean.rb
+++ b/diceBot/Dracurouge_Korean.rb
@@ -87,7 +87,7 @@ MESSAGETEXT
     oneCount = countTargetDice(diceList, 1)
     sixCount = countTargetDice(diceList, 6)
 
-    gloryDiceCount = (oneCount / 2) + (sixCount / 2)
+    gloryDiceCount = oneCount.div(2) + sixCount.div(2)
     return gloryDiceCount
   end
 
@@ -142,7 +142,7 @@ MESSAGETEXT
   type2 = %w{영주	근위	방랑	현자	사냥꾼	야수}
     type2_indexTexts = %w{F G R W J N}
 
-    tensValue = number.to_i / 10
+    tensValue = number.to_i.div(10)
     isBefore = (tensValue < 4)
     type = (isBefore ? type1 : type2)
     indexTexts = (isBefore ? type1_indexTexts : type2_indexTexts)
diff --git a/diceBot/EarthDawn3.rb b/diceBot/EarthDawn3.rb
index a3fce74..642fd83 100644
--- a/diceBot/EarthDawn3.rb
+++ b/diceBot/EarthDawn3.rb
@@ -145,7 +145,7 @@ INFO_MESSAGE_TEXT
 
     result = [  0,   0,   0,  0,  0,  0,   0]
 
-    loopCount = (overStep / stepRythm.size)
+    loopCount = overStep.div(stepRythm.size)
 
     loopCount.times do
       addStepToResult(result, baseStepInfo)
diff --git a/diceBot/EarthDawn4.rb b/diceBot/EarthDawn4.rb
index 16f3f53..cd6fe37 100644
--- a/diceBot/EarthDawn4.rb
+++ b/diceBot/EarthDawn4.rb
@@ -191,7 +191,7 @@ INFO_MESSAGE_TEXT
 
     result = [  0,   0,   0,  0,  0,  0,   0]
 
-    loopCount = (overStep / stepRythm.size)
+    loopCount = overStep.div(stepRythm.size)
 
     loopCount.times do
       addStepToResult(result, overBounusStep)
@@ -223,7 +223,7 @@ INFO_MESSAGE_TEXT
       return "失敗"
     end
 
-    level = (diff / 5) + 1
+    level = diff.div(5) + 1
 
     return "成功 レベル：#{level}"
   end
diff --git a/diceBot/EclipsePhase.rb b/diceBot/EclipsePhase.rb
index 5974a55..e1ea8a0 100644
--- a/diceBot/EclipsePhase.rb
+++ b/diceBot/EclipsePhase.rb
@@ -23,7 +23,7 @@ INFO_MESSAGE_TEXT
     return '' unless signOfInequality == '<='
 
     diceValue = total_n % 100 # 出目00は100ではなく00とする
-    dice_ten_place = diceValue / 10
+    dice_ten_place = diceValue.div(10)
     dice_one_place = diceValue % 10
 
     debug("total_n", total_n)
diff --git a/diceBot/FilledWith.rb b/diceBot/FilledWith.rb
index 661b979..52d7c89 100644
--- a/diceBot/FilledWith.rb
+++ b/diceBot/FilledWith.rb
@@ -1532,7 +1532,7 @@ SLv3:ゴーレムアーム(装飾品)            '『白銀の猛将』ジェネラル(CL55/ネームド)1匹([命中]+4、あらゆる致傷力+20、[HP]+500)'
             '『白銀の猛将』ジェネラル(CL55/ネームド)1匹([命中]+4、あらゆる致傷力+20、[HP]+500)'
          ]],
             ]
-    case event / 100
+    case event.div(100)
       when 1
         areaText = "洞窟"
       when 2
diff --git a/diceBot/GardenOrder.rb b/diceBot/GardenOrder.rb
index 2b495cc..6dfc2db 100644
--- a/diceBot/GardenOrder.rb
+++ b/diceBot/GardenOrder.rb
@@ -50,12 +50,12 @@ INFO_MESSAGE_TEXT
   def get_critical_border(critical_border_text, success_rate)
     return critical_border_text.to_i unless critical_border_text.nil?
 
-    critical_border = [success_rate / 5, 1].max
+    critical_border = [success_rate.div(5), 1].max
     return critical_border
   end
 
   def check_roll_repeat_attack(success_rate, repeat_count, critical_border)
-    success_rate_per_one = success_rate / repeat_count
+    success_rate_per_one = success_rate.div(repeat_count)
 
     check_roll(success_rate_per_one, critical_border)
   end
diff --git a/diceBot/GurpsFW.rb b/diceBot/GurpsFW.rb
index 1285e31..384ca73 100644
--- a/diceBot/GurpsFW.rb
+++ b/diceBot/GurpsFW.rb
@@ -341,7 +341,7 @@ INFO_MESSAGE_TEXT
       tableName = "財宝テーブル"
       diff = $1
       depth = $2.to_i
-      num = depth / 10
+      num = depth.div(10)
       if num >= 6
         num = 5
       end
diff --git a/diceBot/IthaWenUa.rb b/diceBot/IthaWenUa.rb
index eb6db20..ae8d303 100644
--- a/diceBot/IthaWenUa.rb
+++ b/diceBot/IthaWenUa.rb
@@ -23,7 +23,7 @@ MESSAGETEXT
     return '' unless signOfInequality == '<='
 
     diceValue = total_n % 100
-    dice0 = diceValue / 10 #10の位を代入
+    dice0 = diceValue.div(10) #10の位を代入
     dice1 = diceValue % 10 # 1の位を代入
 
     debug("total_n", total_n)
diff --git a/diceBot/LogHorizon.rb b/diceBot/LogHorizon.rb
index ceda84a..b080b09 100644
--- a/diceBot/LogHorizon.rb
+++ b/diceBot/LogHorizon.rb
@@ -480,7 +480,7 @@ MESSAGETEXT
   end
 
   def getTableByRank(rank, tables)
-    index = (rank - 1) / 5
+    index = (rank - 1).div(5)
 
     index = [0, index].max
     index = [index, (tables.size - 1)].min
diff --git a/diceBot/LogHorizon_Korean.rb b/diceBot/LogHorizon_Korean.rb
index 4605fea..b4d25cb 100644
--- a/diceBot/LogHorizon_Korean.rb
+++ b/diceBot/LogHorizon_Korean.rb
@@ -319,7 +319,7 @@ MESSAGETEXT
   end
 
   def getTableByRank(rank, tables)
-    index = (rank - 1) / 5
+    index = (rank - 1).div(5)
 
   index = [0, index].max
   index = [index, (tables.size - 1)].min
diff --git a/diceBot/Ryutama.rb b/diceBot/Ryutama.rb
index 4045ae3..cffe68e 100644
--- a/diceBot/Ryutama.rb
+++ b/diceBot/Ryutama.rb
@@ -84,14 +84,14 @@ INFO_MESSAGE_TEXT
 
     diceBase = dice1
 
-    dice1 = diceBase / 10
+    dice1 = diceBase.div(10)
     dice2 = diceBase % 10

     if isValidDice(dice1, dice2)
       return dice1, dice2
     end
 
-    dice1 = diceBase / 100
+    dice1 = diceBase.div(100)
     dice2 = diceBase % 100
 
     if isValidDice(dice1, dice2)
diff --git a/diceBot/ShinMegamiTenseiKakuseihen.rb b/diceBot/ShinMegamiTenseiKakuseihen.rb
index 0f605be..07a04d2 100644
--- a/diceBot/ShinMegamiTenseiKakuseihen.rb
+++ b/diceBot/ShinMegamiTenseiKakuseihen.rb
@@ -51,7 +51,7 @@ INFO_MESSAGE_TEXT
 
     value %= 100
 
-    dice1 = value / 10
+    dice1 = value.div(10)
     dice2 = value % 10
 
     return [dice1, dice2]
diff --git a/diceBot/Skynauts.rb b/diceBot/Skynauts.rb
index dfbfd3e..767a346 100644
--- a/diceBot/Skynauts.rb
+++ b/diceBot/Skynauts.rb
@@ -97,7 +97,7 @@ MESSAGETEXT
     debug("移動修正", bonus)
 
     total, = roll(1, 6)
-    movePointBase = (total / 2) <= 0 ? 1 : (total / 2)
+    movePointBase = total.div(2) <= 0 ? 1 : total.div(2)
     movePoint = movePointBase + bonus
     debug("移動エリア数", movePoint)
 
diff --git a/diceBot/Warhammer.rb b/diceBot/Warhammer.rb
index 504e1d7..ad73bc1 100644
--- a/diceBot/Warhammer.rb
+++ b/diceBot/Warhammer.rb
@@ -53,10 +53,10 @@ INFO_MESSAGE_TEXT
     return '' unless signOfInequality == "<="
 
     if total_n <= diff
-      return " ＞ 成功(成功度#{ ((diff - total_n) / 10) })"
+      return " ＞ 成功(成功度#{ (diff - total_n).div(10) })"
     end
 
-    return " ＞ 失敗(失敗度#{ ((total_n - diff) / 10) })"
+    return " ＞ 失敗(失敗度#{ (total_n - diff).div(10) })"
   end
 
 ####################            WHFRP関連          ########################
