diff --git a/bcdiceCore.rb b/bcdiceCore.rb
index a8324da..4b75c97 100644
--- a/bcdiceCore.rb
+++ b/bcdiceCore.rb
@@ -176,7 +176,7 @@ class BCDice
   end
 
   def printErrorMessage(e)
-    sendMessageToOnlySender("error " + e.to_s + $@.join("\n"))
+    sendMessageToOnlySender("error " + e.to_s + ($@ || []).join("\n") + $!)
   end
 
   def recieveMessageCatched(nick_e, tnick)
diff --git a/diceBot/DetatokoSaga.rb b/diceBot/DetatokoSaga.rb
index dec7207..bd452f5 100644
--- a/diceBot/DetatokoSaga.rb
+++ b/diceBot/DetatokoSaga.rb
@@ -155,6 +155,7 @@ INFO_MESSAGE_TEXT
   end
 
   def getModifyText(operator, value)
+    return '' if( value == 0)
     operatorText =
       case operator
       when "+"
diff --git a/diceBot/DetatokoSaga_Korean.rb b/diceBot/DetatokoSaga_Korean.rb
index 5580409..b7dfdaa 100644
--- a/diceBot/DetatokoSaga_Korean.rb
+++ b/diceBot/DetatokoSaga_Korean.rb
@@ -155,6 +155,7 @@ INFO_MESSAGE_TEXT
   end
 
   def getModifyText(operator, value)
+    return '' if( value == 0)
     operatorText =
       case operator
       when "+"
diff --git a/diceBot/DiceBot.rb b/diceBot/DiceBot.rb
index 109d4bf..44385af 100644
--- a/diceBot/DiceBot.rb
+++ b/diceBot/DiceBot.rb
@@ -234,7 +234,7 @@ class DiceBot
       debug('call rollDiceCommand command', command)
       result, secret_flg = rollDiceCommand(command)
     rescue => e
-      debug("executeCommand exception", e.to_s, $@.join("\n"))
+      debug("executeCommand exception", e.to_s, ($@ || []).join("\n"), $!)
     end
 
     debug('rollDiceCommand result', result)
diff --git a/diceBot/MetalHeadExtream.rb b/diceBot/MetalHeadExtream.rb
index 4b2839c..a18f312 100644
--- a/diceBot/MetalHeadExtream.rb
+++ b/diceBot/MetalHeadExtream.rb
@@ -908,7 +908,7 @@ MESSAGETEXT
   # 端数が使いたいので、parren_killer未使用
   def get_value(originalValue, calculateText)
     result = originalValue.to_f
-    calculateArray = calculateText.scan(/[\*\/]\d*/)  # 修正値を分割して配列へ
+    calculateArray = (calculateText || '').scan(/[\*\/]\d*/) # 修正値を分割して配列へ
     calculateArray.each do |i|                        # 配列内ループで補正率を計算
       i =~ /([\*\/])(\d*)/i
       result *= $2.to_i if($1 == '*')
diff --git a/diceBot/Postman.rb b/diceBot/Postman.rb
index e741935..89b10f7 100644
--- a/diceBot/Postman.rb
+++ b/diceBot/Postman.rb
@@ -55,11 +55,11 @@ MESSAGETEXT
         diceCount = 2 if(diceCount < 2)
 
         modify = ($2 || 0).to_i
-        modifyAddString = $3
+        modifyAddString = ($3 || '')
 
         type = ($6 || '')
         target = ($7 || 0).to_i
-        targetAddString = $8
+        targetAddString = ($8 || '')
 
         modify_list = modifyAddString.scan(/[+-]\d+/)
         modify_list.each{|i| modify += i.to_i }
diff --git a/irc/ircLib.rb b/irc/ircLib.rb
index 595dd7d..9603e75 100644
--- a/irc/ircLib.rb
+++ b/irc/ircLib.rb
@@ -25,7 +25,7 @@ class Net::IRC::Client
       @log.debug "error!!";
       @log.debug e.to_s
       @log.debug $!.inspect
-      @log.debug $@.join("\n")
+      @log.debug $@&.join("\n")
     end
     @log.debug "on_connected passed";
 
