diff --git a/diceBot/DiceBotLoader.rb b/diceBot/DiceBotLoader.rb
index 3765518..84a02ec 100644
--- a/diceBot/DiceBotLoader.rb
+++ b/diceBot/DiceBotLoader.rb
@@ -229,9 +229,8 @@ class DiceBotLoader
   # @return [DiceBot]
   def loadDiceBot
     @filenames.each do |filename|
-      require(
-        File.expand_path(filename, File.dirname(__FILE__))
-      )
+      require_path = File.expand_path(filename, File.dirname(__FILE__))
+      require(require_path)
     end
 
     Object.const_get(@diceBotClass).new
diff --git a/diceBot/MetalHeadExtream.rb b/diceBot/MetalHeadExtream.rb
index 4cd46a1..4b2839c 100644
--- a/diceBot/MetalHeadExtream.rb
+++ b/diceBot/MetalHeadExtream.rb
@@ -587,7 +587,7 @@ MESSAGETEXT
                    ['H', '(HW)'],
                    ['O', '(MO)']]
 
-    index = damageInfos.index { |type, text| type == damageStage }
+    index = damageInfos.index { |type| type.first == damageStage }
     return nil if index == -1
 
     damageIndex = index + 1
